nixos/modules/services/cluster/patroni/default.nix

```
@@ -0,0 +1,249 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+  cfg = config.services.patroni;
+  format = pkgs.formats.yaml { };
+
+  #boto doesn't support python 3.10 yet
+  patroni = pkgs.patroni.override { pythonPackages = pkgs.python39Packages; };
+
+  postgresql =
+    if cfg.postgresqlExtraPlugins == [ ]
+    then cfg.postgresqlPackage
+    else cfg.postgresqlPackage.withPackages (_: cfg.postgresqlExtraPlugins);
+
+  configFile = format.generate "patroni.yml" cfg.settings;
+in
+{
+  options.services.patroni = {
+
+    enable = mkEnableOption "Whether to enable Patroni";
+
+    postgresqlPackage = mkOption {
+      type = types.package;
+      example = literalExpression "pkgs.postgresql_11";
+      description = ''
+        PostgreSQL package to use.
+      '';
+    };
+
+    postgresqlExtraPlugins = mkOption {
+      type = types.listOf types.path;
+      default = [ ];
+      example = literalExpression "with pkgs.postgresql_11.pkgs; [ postgis pg_repack ]";
+      description = ''
+        List of PostgreSQL plugins. PostgreSQL version for each plugin should
+        match version for <literal>services.postgresql.package</literal> value.
+      '';
```

```suggestion
      description = mdDoc ''
        List of PostgreSQL plugins. PostgreSQL version for each plugin should
        match version for `services.postgresql.package` value.
      '';
```
module docs should preferentially be written in markdown syntax, and ideally all docs should be marked as markdown even if they don't contain docbook tags

---

Will do.